### ISSUE TITLE
Cross-validation during training

### ISSUE THREAD

**Issue by aminekechaou** [RasaHQ/rasa#6437]
It would be nice to have cross-validation done periodically during training in order to know when to early-stop and avoid overfitting. A useful feature would be to allow the user to define a validation set, in a similar way as the training data, on which to perform the evaluation periodically.

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @rgstephens will get back to you about it soon! ✨

**Comment by rgstephens** [comment]
The `rasa test --cross-validation` allows you to pass test data via the `--nlu` option.  One problem with automatically doing cross-validation during training is the fact that training is part of CI/CD pipelines and can take significant resources.

**Comment by aminekechaou** [comment]
For anyone looking for this feature if you're using DIETClassifier, you can set the evaluate_every_number_of_epochs and evaluate_on_number_of_examples  parameters under the DIETClassifier component in config.yml
